### PR TITLE
Optimizations + memory leak fix

### PR DESCRIPTION
**Pull request by devdani1**
We encountered a memory leak issue on our server.

The problem got fixed after adding this line: `lastMovementTimes.remove(player);` 
The memory leak occurred because players would be kept on the map even after disconnecting.

I also made some minor optimizations.

### CODE REVIEW

File: src/main/java/me/serbob/toastedafk/Events/Custom/Listener/RegionRelatedEventsHandler.java [hunk 1]
```diff
@@ -12,38 +12,29 @@
...
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
 
 import static me.serbob.toastedafk.Managers.ValuesManager.*;
 import static me.serbob.toastedafk.Templates.CoreHelpers.removePlayer;
 
 public class RegionRelatedEventsHandler implements Listener {
-    @EventHandler
```
Comment (devdani1) [inline src/main/java/me/serbob/toastedafk/Events/Custom/Listener/RegionRelatedEventsHandler.java:22]: PlayerQuitEvent is always called after this event, so this is not needed

**Comment by SerbanHiro** [comment]
Seems good, but I do have 1 question, you've changed from ConcurrentHashMap to HashMap because PlayerMoveEvent isn't threaded, it's on the server's main thread?

**Comment by devdani1** [comment]
Yes, since it is called only from the main thread, it doesn't need to be concurrent.

**Comment by SerbanHiro** [comment]
Got it, just wanted to make sure.

**Pull request merged**